app/models/miq_request_workflow.rb

```diff
@@ -1042,6 +1042,9 @@ def allowed_storages(_options = {})
     storages = hosts.each_with_object({}) do |host, hash|
       host.writable_storages.each { |s| hash[s.id] = s }
     end.values
+
+    storages.reject!(&:maintenance)
```

Comment:
We could add a `Host::available_storages` method that combines the maintenance mode check and the writable check.  There is a `read_only_storages` method that checks for `:read_only => true` and I'd expect `writable_storages` to be the inverse, not include another check for `maintenance`.